### DISCUSSION TITLE
Is there an "fill(previous)" option in Prometheus like in Flux?

### DISCUSSION THREAD

**Discussion by timoth3us** [prometheus/prometheus#10950] [Q&A]
Prometheus Version: v2.28.1

Hi,
I use Prometheus as a data source for Grafana. The data is is coming from an opc-ua subscription model, which is why it is very sparse sometimes. However, there is a need to extend the last value of the timeseries to the current moment, to show a full time range in grafana. In Flux there is a function that does exactly that, that is called "fill(previous)". Is there anything like it in Prometheus?

Thanks in Advance!

**Comment by roidelapluie** [comment]
You might try last_over_time(my_metric[30m])
